pkg/components/exec_node_base_test.go

```diff
@@ -0,0 +1,320 @@
+					Format:      ytv1.LogFormatPlainText,
+					MinLogLevel: ytv1.LogLevelDebug,
+					Compression: ytv1.LogCompressionZstd,
+					RotationPolicy: &ytv1.LogRotationPolicy{
+						RotationPeriodMilliseconds: &rotationPolicyMS,
+						MaxTotalSizeToKeep:         &rotationPolicyMaxTotalSize,
+					},
+				},
+				WriterType: ytv1.LogWriterTypeFile,
+				CategoriesFilter: &ytv1.CategoriesFilter{
+					Type:   ytv1.CategoriesFilterTypeExclude,
+					Values: []string{"Bus", "Concurrency"},
+				},
+			},
+		},
+		Name: "end-a",
+	}
+}
+
+func withCri(spec ytv1.ExecNodesSpec, jobResources *corev1.ResourceRequirements, isolated bool) ytv1.ExecNodesSpec {
+	spec.Locations = append(spec.Locations, testLocationImageCache)
+	spec.JobResources = jobResources
+	spec.JobEnvironment = &ytv1.JobEnvironmentSpec{
+		UserSlots: ptr.Int(42),
+		CRI: &ytv1.CRIJobEnvironmentSpec{
+			SandboxImage:           ptr.String("registry.k8s.io/pause:3.8"),
+			APIRetryTimeoutSeconds: ptr.Int32(120),
+			CRINamespace:           ptr.String("yt"),
+			BaseCgroup:             ptr.String("/yt"),
+		},
+		UseArtifactBinds: ptr.Bool(true),
+		DoNotSetUserId:   ptr.Bool(true),
+		Isolated:         ptr.Bool(isolated),
+	}
+	return spec
+}
+
+func TestDoBuildExecNodeBase(t *testing.T) {
+	yt := getYtsaurus()
+	ytsaurus := apiProxy.NewYtsaurus(yt, nil, nil, nil)
+
+	l := labeller.Labeller{
+		ObjectMeta:     ptr.T(ytsaurus.GetResource().ObjectMeta),
+		APIProxy:       ytsaurus.APIProxy(),
+		ComponentLabel: consts.YTComponentLabelExecNode,
+		ComponentName:  consts.YTComponentLabelExecNode,
+	}
+
+	g := ytconfig.NewLocalNodeGenerator(yt, testClusterDomain)
+
+	cases := map[string]struct {
+		JobResources *corev1.ResourceRequirements
+		Isolated     bool
+	}{
+		"isolated-containers-without-job-resources": {
+			JobResources: nil,
+			Isolated:     true,
+		},
+		"isolated-containers-with-job-resources": {
+			JobResources: &testJobResourceReqs,
+			Isolated:     true,
+		},
+		"single-container-without-job-resources": {
+			JobResources: nil,
+			Isolated:     false,
+		},
+		"single-container-with-job-resources": {
+			JobResources: &testJobResourceReqs,
+			Isolated:     false,
+		},
+	}
+
+	for name, test := range cases {
+		t.Run(name, func(t *testing.T) {
+			execNodeSpec := withCri(getExecNodeSpec(nil), test.JobResources, test.Isolated)
+
+			srv := newServer(
+				&l,
+				ytsaurus,
+				ptr.T(execNodeSpec.InstanceSpec),
+				"/usr/bin/ytserver-node",
+				"ytserver-exec-node.yson",
+				consts.YTComponentLabelExecNode,
+				consts.YTComponentLabelExecNode,
+				func() ([]byte, error) {
+					return g.GetExecNodeConfig(execNodeSpec)
+				},
+				WithContainerPorts(corev1.ContainerPort{
+					Name:          consts.YTRPCPortName,
+					ContainerPort: consts.ExecNodeRPCPort,
+					Protocol:      corev1.ProtocolTCP,
+				}),
+			)
+
+			bEN := baseExecNode{
+				server:        srv,
+				cfgen:         g,
+				spec:          &execNodeSpec,
+				sidecarConfig: nil,
+			}
```

Comment:
Also this test looks rather fragile since it tests low-level implementation of the component and will often broke on refactorings. Does it really test something that couldn't be tested in ytconfig tests?

Reply:
I've reverted this particular test file https://github.com/ytsaurus/yt-k8s-operator/pull/281
Could you please send another PR with test if you feel it is needed (I expect them to be among `ytconfig` tests and without manipulating base struct)?